gridstatus/ieso.py

```diff
@@ -997,7 +997,8 @@ def _get_latest_resource_adequacy_json(
             )
 
         if last_modified:
-            last_modified = pd.Timestamp(last_modified, tz=self.default_timezone)
+            if last_modified.tz is None:
+                last_modified = last_modified.tz_localize("UTC")
```

Comment:
I don't understand the conversion to UTC. Aren't the last modified times from [this page](https://reports-public.ieso.ca/public/Adequacy2/?C=M;O=D) in EST? 

`date` and `end` are converted to EST so it would be inconsistent to convert `last_modified` to UTC.

Reply:
To clarify, when a function is decorated with `@support_date_range`, the `date` and `end` parameters get converted to the local timezone using the `utils._handle_date` method. So, maybe we should pass `last_modified` through `utils._handle_date`.

https://github.com/gridstatus/gridstatus/blob/88acf30dcefa0e7c6941ac5fc04193fa83c41718/gridstatus/decorators.py#L95-L108

Reply:
Yeah mostly I was converting to UTC here to make it unambiguous and explicit. The issue that was occurring was that a (timezone-naive) UTC time was being given to the kwarg `last_modified` which was then being (incorrectly) localized to local time here, resulting in no data. 

So while yes the last modified times from the page are in EST, the date comparison used to filter out the files can compare two timestamps in different timezones correctly, as long as there is a known timezone to each. (If one is timezone-naive and the other is timezone-aware, the date comparison will error out with the classic `Error: Cannot convert tz-naive Timestamp, use tz_localize to localize`, which is why the localization was added here in the first place). But if the naive time is actually in UTC, then localizing to EST is not correct. 

So the options are:
1. Expect a timezone-aware time, error if not
2. Handle a timezone-naive time to something (but only if it is naive). 

I opt for 2, but can change that handling to EST rather than UTC (perhaps using `utils._handle_date`).

Reply:
> I opt for 2, but can change that handling to EST rather than UTC (perhaps using utils._handle_date).

this would be consistent with the rest of the library.